### ISSUE TITLE
Typescript package dependency

### ISSUE THREAD

**Issue by alexdrel** [jbrantly/ts-loader#28]
Hello, can you please consider removing typescript from package "dependencies" and adding it to peerDependencies and devDependencies? 
The "compiler" option allows to switch to ntypescript but not typescript@next, ts-loader@^0.4 works fine after I manually moved the dependency.  

```json
  "peerDependencies": {
    "typescript": ">=1.5.3"
  },
  "devDependencies": {
    "typescript": ">=1.5.3"
  }
```

The only drawback is that with npm@3 (https://github.com/npm/npm/issues/6565) the peerDependencies will not be installed automatically but IMHO returning to the host project control over the typescript version is more important. 

Thanks for the loader!

**Comment by jbrantly** [comment]
Yup, I complete agree, which is why I already made the change in d51ecae :) I just haven't published to NPM yet because I've got a few more things I'd like to squeeze into the release. I'm planning to push to NPM either tomorrow or Monday.

I'm not planning on adding to peerDependencies because that would be strange with something like ntypescript or jsx-typescript, where you really don't want the official typescript in the first place. I'm erring on the side of flexibility over convenience. 

Also, going forward I'm planning on keeping ts-loader completely compatible with typescript@next via [extensive nightly tests](https://travis-ci.org/jbrantly/ts-loader).

**Comment by alexdrel** [comment]
Great, thanks :+1: 

I think you could have leave it as a devDependency to simplify your process but of course it is up to you.

Although I agree with you regarding peerDependencies I feel that going forward the "compiler" switch will not see much use, with jsx-ts fork outdated (it was a live saver while it lasted) and ntypescript naming complicates integration with sublime/vs code.

**Comment by jbrantly** [comment]
BTW, this was published in v0.5.0
